feat(ast): remove explicit untagged marker on enums

ottomated opened the pull request:
This assumes that any enums with exactly one field in each variant are untagged, and they're tagged otherwise.

ottomated commented:
* **#6915** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6915?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ottomated and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

overlookmotel reviewed (approved):
Great!

overlookmotel commented:
### Merge activity

* **Oct 26, 4:21 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

ottomated committed:
feat(ast): remove explicit untagged marker on enums (#6915)

This assumes that any enums with exactly one field in each variant are untagged, and they're tagged otherwise.
